libfwupdplugin/fu-device.c

```diff
@@ -3228,9 +3240,10 @@ fu_device_finalize (GObject *object)
 		g_object_unref (priv->quirks);
 	if (priv->poll_id != 0)
 		g_source_remove (priv->poll_id);
+	if (priv->metadata != NULL)
+		g_hash_table_unref (priv->metadata);
 	g_rw_lock_clear (&priv->metadata_mutex);
 	g_rw_lock_clear (&priv->parent_guids_mutex);
-	g_hash_table_unref (priv->metadata);
 	g_ptr_array_unref (priv->children);
 	g_ptr_array_unref (priv->parent_guids);
```

Comment:
Perhaps this should also move up above the mutex release too?

Reply:
I think we should clear the mutex before we finalize the thing it protects.

Reply:
OK.

Reply:
I guess the other one needs to be inverted then again too.

Reply:
Fixed up.